controllers/dependencies.go

```diff
@@ -0,0 +1,170 @@
+package controllers
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"os"
+
+	"path/filepath"
+
+	git "github.com/go-git/go-git/v5"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	OPERATOR_PIPELINES_REPO = "https://github.com/redhat-openshift-ecosystem/operator-pipelines.git"
+	REPO_CLONE_PATH         = "/tmp/operator-pipelines/"
+	PIPELINE_MANIFESTS_PATH = "ansible/roles/operator-pipeline/templates/openshift/pipelines"
+	TASKS_MANIFESTS_PATH    = "ansible/roles/operator-pipeline/templates/openshift/tasks"
+)
+
+func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(meta metav1.ObjectMeta) error {
+
+	// Cloning operator-pipelines project to retrieve pipelines and tasks
+	// yaml manifests that need to be applied beforehand
+	// ref: https://github.com/redhat-openshift-ecosystem/certification-releases/blob/main/4.9/ga/ci-pipeline.md#step-6---install-the-certification-pipeline-and-dependencies-into-the-cluster
+
+	_, err := git.PlainClone(REPO_CLONE_PATH, false, &git.CloneOptions{
+		URL:      OPERATOR_PIPELINES_REPO,
+		Progress: os.Stdout,
+	})
+	if err != nil {
+		log.Log.Info("Couldn't clone the repository for operator-pipelines.")
+		return err
+	}
+	defer r.RemovePipelineDependencyFiles(REPO_CLONE_PATH)
+
+	// Reading pipeline manifests and applying to cluster
+	root := REPO_CLONE_PATH + PIPELINE_MANIFESTS_PATH
```

Comment:
These two blocks could be a loop.

Reply:
Can you clarify? What do you mean?

Reply:
paths := []string{PIPELINE_MANIFESTS_PATH, TASKS_MANIFESTS_PATH}
for _, path := range paths {
   ...
}

The blocks are identical. The only difference is the two paths.

Reply:
Done.